core/constraint-streams/src/main/java/ai/timefold/solver/constraint/streams/bavet/common/AbstractFlattenLastNode.java

```diff
@@ -103,6 +123,29 @@ public void update(InTuple_ tuple) {
         }
     }
 
+    private List<FlattenedItem_> iterableToList(Iterable<FlattenedItem_> iterable) {
+        if (iterable instanceof Collection<FlattenedItem_> collection) {
+            // Optimization for Collection, where we know the size.
+            int size = collection.size();
+            if (size == 0) {
+                return Collections.emptyList();
+            }
+            List<FlattenedItem_> result = new ArrayList<>(size);
```

Comment:
In this case, it is safe to cast the `iterable` to a `Collection` and pass it here directly. It could bring an additional minor perf improvement, if we skip the `forEach` and the method reference (no proof).

Reply:
I actually tried that and, for some reason, sizing the collection and filling it like this is marginally *faster* than initializing the list with the collection directly. Overall, the difference wasn't big enough for me to pursue this further.